src/components/App.js

```diff
@@ -1,18 +1,199 @@
-import React from 'react';
+import React from "react";
+import OrderMenu from "./OrderMenu";
+import Order from "./Order";
+import MenuPage from "./MenuPage";
+import Navigation from "./Navigation";
+import Header from "./Header";
+import cx from "classnames";
 
-import '../styles/App.scss';
+import "../styles/App.scss";
 
 class App extends React.Component {
-  constructor(){
+  constructor() {
     super();
+    this.fetchMenu = this.fetchMenu.bind(this);
+    this.receiveOrder = this.receiveOrder.bind(this);
+    this.receiveSubmitOrder = this.receiveSubmitOrder.bind(this);
+    this.receiveQuanitityIncrease = this.receiveQuanitityIncrease.bind(this);
+    this.receiveQuanitityDecrease = this.receiveQuanitityDecrease.bind(this);
+    this.updateTotalPrice = this.updateTotalPrice.bind(this);
+    this.checkCurrentOrder = this.checkCurrentOrder.bind(this);
+    this.formatToMoney = this.formatToMoney.bind(this);
+    this.receivePageChange = this.receivePageChange.bind(this);
+    this.state = {
+      menuObject: {
+        burgers: [],
+        fries: [],
+        extras: []
+      },
+      currentOrder: { orderItems: [], orderTotal: 0 },
+      previousOrders: [],
+      currentPage: "menu"
+    };
   }
 
-  render(){
+  receivePageChange(id) {
+    if (id === "menu") {
+      this.setState({
+        currentPage: "menu"
+      });
+    } else if (id === "order") {
+      this.setState({
+        currentPage: "order"
+      });
+    }
+  }
+
+  componentDidMount() {
+    this.fetchMenu();
+    this.setState({
+      currentOrder: 
+      localStorage.getItem("currentOrder") === null
+            ? { orderItems: [], orderTotal: 0 }
+            : JSON.parse(localStorage.getItem("currentOrder"))
+    })
+  }
+
+  fetchMenu() {
+    fetch("/api/menu")
+      .then(response => response.json())
+      .then(body =>
+        this.setState({
+          menuObject: {
+            burgers: Object.values(body.burgers),
+            fries: Object.values(body.fries),
+            extras: Object.values(body.extras)
+          }
+        })
+      );
+  }
+
+  formatToMoney(number) {
+    return number.toLocaleString("en-GB", {
+      style: "currency",
+      currency: "GBP"
+    });
+  }
+
+  updateTotalPrice(orderList) {
+    let orderPriceArray = [];
+    orderList.map(order => {
+      orderPriceArray.push(order.price * order.quantity);
+    });
+    return orderPriceArray.reduce((acc, total) => acc + total);
+  }
+
+  receiveOrder(order) {
+    order.quantity = 1;
+    if (this.state.currentOrder.orderItems.includes(order) === false) {
+      const incomingOrder = this.state.currentOrder.orderItems.concat(order);
+      const totalPrice = this.updateTotalPrice(incomingOrder);
+      this.setState(
+        {
+          currentOrder: {
+            orderItems: incomingOrder,
+            orderTotal: totalPrice
+          }
+        },
+        () => localStorage.setItem("currentOrder", JSON.stringify(this.state.currentOrder))
```

Comment:
The callback can be avoided here, passing `incomingOrder` to localStorage rather than waiting for setState to complete and inserting data into localStorage after